src/System/Ctags/Internal.hs

```diff
@@ -13,25 +20,47 @@ import System.Ctags.IO
 import System.Ctags.Types
 import qualified System.Directory as D
 
-tagsContent :: MonadIO m => m (Either TagSearchOutcome T.Text)
-tagsContent = readTagsFile =<< findTagsFile
+type AppConfig = MonadReader CtagsSearchConfig
+
+newtype App a = App
+    { runApp :: ReaderT CtagsSearchConfig (ExceptT TagSearchOutcome IO) a
+    } deriving ( Functor
+               , Applicative
+               , Monad
+               , AppConfig
+               , MonadIO
+               , MonadError TagSearchOutcome
+               )
+
+data CtagsSearchConfig = CtagsSearchConfig
+    { tagFileName :: String
+    , tagFileDirectories :: [String]
+    }
+
+defaultCtagsSearchConfig :: CtagsSearchConfig
```

Comment:
There is a type class for this if you'd like to use it: https://hackage.haskell.org/package/data-default-0.7.1.1/docs/Data-Default.html